minor patches related to rrt and shortcutting

v4hn opened the pull request:
Hi Sebastian,

I just used pyroboplan for some visualization and found a few things I would have preferred differently in the beginning.
I guess your defaults reflect your own preferences, so feel free to discard individual commits.

58af2853de0177c2d47c9cb005defb60f776fda0 and be6ec2a3a9aaae6e14a49faa3bf4e76c70a2cde9 fix relatively common hangups with shortcutting, so I also enabled it by default now that it works reliably (for me).

Great work! This was the first time I looked at part of the code.

sea-bass commented on examples/optimize_rrt_path.py:
```diff
@@ -116,16 +116,17 @@ def random_valid_state():
-                show_jerk=True,
-            )
+            # plot the trajectory and points along the path.
+            if False:
```
Putting an `if False` seems a bit strange. Maybe just make a boolean variable?

v4hn replied:
I assume you meant to introduce a variable and directly use it, similar to how you do it in other places.

sea-bass commented on examples/rrt_panda.py:
```diff
@@ -43,11 +43,11 @@
         max_connection_dist=0.5,
-        rrt_connect=False,
-        bidirectional_rrt=False,
+        rrt_connect=True,
```
I would like to keep the defaults as they are in this file. The intention is for people to start with vanilla RRT, see its issues, and explore improvements themselves.

sea-bass commented on src/pyroboplan/planning/path_shortcutting.py:
```diff
@@ -121,18 +121,19 @@ def shortcut_path(
 
             # Sample two points along the path length
             path_scalings = get_normalized_path_scaling(q_shortened)
-            while True:
```
Yeah, why did I have a while loop here? Thank you.

sea-bass reviewed (commented):
Thanks for looking at this and making improvements! Please see my comments here.

v4hn commented:
I addressed your feedback and fixed another edge-condition that was broken (`RRTPlanner.plan` missed the goal waypoint when direct connection was possible).

sea-bass commented on examples/rrt_panda.py:
```diff
@@ -61,27 +61,27 @@
 
             discretized_path = discretize_joint_space_path(path, options.max_step_size)
 
-            if do_shortcutting:
```
I think this if statement was here too because if shortcutting is false, then there should be no "shortened_path" visualization. As is now, if you have shortcutting disabled, the visualization will still be named "shortened_path" which could be misleading.

Additionally, if shortcutting is enabled, I also think you would not get to see the pre-shortcutting path, right? It would only plot the RRT and then the shortcut path. The idea is to have both paths available so you can toggle them on/off in Meshcat to compare.

v4hn replied:
I renamed it to `planned_path` to avoid the misleading name.

sea-bass replied:
I'll put it back to how it was because we actually want the ability to visualize both `planned_path` (pre-shortcutting) and `shortcutted_path` (post-shortcutting) in the same run.

v4hn commented on examples/rrt_panda.py:
```diff
@@ -61,27 +61,27 @@
         if path:
             # Optionally shortcut the path
-            do_shortcutting = False
+            do_shortcutting = True
```
The downside with keeping the defaults all "naive" is that many users will probably not play with it long enough to find the good options and only remember their first impression.
A colleague who worked with robots and planned paths for many years just passed by my screen the other day and asked me what that visualization (of an RRT-Connect tree) was because there were so few edges.
They expected every tree generated by RRT-variants would look like densely filled original RRT, but RRT-Connect trees are almost always sparse.


I changed the default again as you asked.
